mne/utils/mixin.py

```diff
@@ -15,6 +16,14 @@
 from .check import _check_pandas_installed, _check_preload, _validate_type
 from .numerics import _time_mask, object_hash, object_size
 
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing import TypeVar
+
+    Self = TypeVar("Self")
```

Comment:
I already clicked the merge button, but now I'm wondering if we want to start collecting bits like this into something like `mne/utils/typing.py` or similar, for easier re-use across the package.

Reply:
This might be a good idea, yes!